src/Resolvers/DataValidationRulesResolver.php

```diff
@@ -223,7 +224,7 @@ protected function resolveToplevelRules(
     }
 
 
-    protected function resolveOverwrittenRules(
+    protected function resolveRules(
```

Comment:
Let us rename this to appendOverwrittenRules, resolveRules is too general.

Reply:
I've reverted this back to `resolveOverwrittenRules` - I agree `resolveRules` is too generic, but `appendOverwrittenRules` could suggest this method is always merging rather than replacing.

Let me know what you think :)